resolver/type_syntax.cc

```diff
@@ -673,7 +673,7 @@ TypeSyntax::ResultType interpretTCombinator(core::Context ctx, const ast::Send &
                 }
             }
 
-            if (!ctx.owner.isSingletonClass(ctx)) {
+            if (!ctx.owner.isClassOrModule() || !ctx.owner.asClassOrModuleRef().data(ctx)->isSingletonClass(ctx)) {
```

Comment:
This is a little verbose, though I guess if you already have a `ClassOrModuleRef` it's not so bad.